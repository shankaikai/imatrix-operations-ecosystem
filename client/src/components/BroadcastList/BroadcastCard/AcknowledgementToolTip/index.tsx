import React from "react";

interface AcknowledgementToolTipProps {
  AIFS: BroadcastRecipient[];
}

import { useState } from "react";
import {
  Popover,
  Badge,
  Image,
  Text,
  Stack,
  Group,
  Avatar,
  Indicator,
} from "@mantine/core";
import { BroadcastRecipient } from "../../../../proto/operations_ecosys_pb";

export default function AcknowledgementToolTip() {
  const [opened, setOpened] = useState(false);

  return (
    <Popover
      opened={opened}
      onClose={() => setOpened(false)}
      position="bottom"
      placement="end"
      withArrow
      trapFocus={false}
      closeOnEscape={false}
      transition="pop-top-left"
      styles={{ body: { pointerEvents: "none" } }}
      target={
        <Badge
          onMouseEnter={() => setOpened(true)}
          onMouseLeave={() => setOpened(false)}
          variant="filled"
          color="gray"
        >
          AIFS 1
        </Badge>
      }
    >
      <Stack>
        <Group position="apart">
          <Text size="sm">AIFS 1</Text>
          <Text size="sm">Location</Text>
        </Group>
        <Group>
          <Stack align="center" spacing="xs">
            <Indicator color="green" offset={5} withBorder size={14}>
              <Avatar radius="xl" />
            </Indicator>
            <Text size="xs">Name</Text>
            <Text size="xs">HP: 91119999</Text>
          </Stack>
          <Stack align="center" spacing="xs">
            <Indicator color="green" offset={5} withBorder size={14}>
              <Avatar radius="xl" />
            </Indicator>
            <Text size="xs">Name</Text>
            <Text size="xs">HP: 91119999</Text>
          </Stack>
          <Stack align="center" spacing="xs">
            <Indicator color="green" offset={5} withBorder size={14}>
              <Avatar
                radius="xl"
                src="https://images.unsplash.com/photo-1535713875002-d1d0cf377fde?ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=250&q=80"
              />
            </Indicator>
            <Text size="xs">Name</Text>
            <Text size="xs">HP: 91119999</Text>
          </Stack>
        </Group>
      </Stack>
    </Popover>
  );
}
