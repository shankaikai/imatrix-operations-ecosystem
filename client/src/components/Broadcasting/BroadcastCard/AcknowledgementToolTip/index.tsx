import {
  Avatar,
  Badge,
  Group,
  Indicator,
  Popover,
  Stack,
  Text,
} from "@mantine/core";
import React, { useState } from "react";
import { AIFSRecipient } from "../../../../helpers/recipientsFormatter";

interface AcknowledgementToolTipProps {
  data: AIFSRecipient;
}

export default function AcknowledgementToolTip({
  data,
}: AcknowledgementToolTipProps) {
  const [opened, setOpened] = useState(false);

  return (
    <Popover
      opened={opened}
      onClose={() => setOpened(false)}
      position="bottom"
      placement="end"
      withArrow
      trapFocus={false}
      closeOnEscape={false}
      transition="pop-top-left"
      styles={{ body: { pointerEvents: "none" } }}
      target={
        <Badge
          onMouseEnter={() => setOpened(true)}
          onMouseLeave={() => setOpened(false)}
          variant="filled"
          color={data.allAcknowledged ? "green" : "gray"}
        >
          {data.id}
        </Badge>
      }
    >
      <Stack>
        <Group position="apart">
          <Text size="sm">{data.id}</Text>
          <Text size="sm">{data.location}</Text>
        </Group>
        <Group>
          {data.users.map((user) => (
            <Stack key={user.id} align="center" spacing="xs">
              <Indicator
                color={user.acknowledged ? "green" : "red"}
                offset={5}
                withBorder
                size={14}
              >
                <Avatar radius="xl" src={user.img} />
              </Indicator>
              <Text size="xs">{user.name}</Text>
              <Text size="xs">HP: {user.phone}</Text>
            </Stack>
          ))}
        </Group>
      </Stack>
    </Popover>
  );
}
