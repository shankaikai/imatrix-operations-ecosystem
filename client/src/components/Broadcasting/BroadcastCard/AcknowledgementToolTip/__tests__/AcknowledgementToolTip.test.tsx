import { render, screen } from "@testing-library/react";
import AcknowledgementToolTip from "..";
import {
  AIFSRecipient,
  RecipientDetails,
} from "../../../../../helpers/recipientsFormatter";

const mockRecipientDetails: RecipientDetails[] = [
  {
    id: 1,
    phone: "some phone number",
    acknowledged: true,
    img: "some url",
    name: "some name",
  },
];

const mockData: AIFSRecipient = {
  id: "some id",
  location: "some location",
  allAcknowledged: true,
  users: mockRecipientDetails,
};

describe("AcknowledgementToolTip", () => {
  it("matches snapshot", () => {
    const { container } = render(<AcknowledgementToolTip data={mockData} />);
    expect(container).toMatchSnapshot();
  });
});
