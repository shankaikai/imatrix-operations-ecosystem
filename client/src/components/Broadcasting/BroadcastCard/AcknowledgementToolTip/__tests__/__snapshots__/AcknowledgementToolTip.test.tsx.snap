// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AcknowledgementToolTip matches snapshot 1`] = `
<div>
  <div
    class="mantine-Popover-root mantine-gc2cx1"
  >
    <div
      class="mantine-1imxcuq mantine-Popover-target"
    >
      <div
        class="mantine-Badge-filled mantine-Badge-root mantine-12ynvk6"
      >
        <span
          class="mantine-h9iq4m mantine-Badge-inner"
        >
          some id
        </span>
      </div>
    </div>
  </div>
</div>
`;
